std/zig/parser_test.zig

```diff
@@ -2246,6 +2246,76 @@ test "zig fmt: if type expr" {
     );
 }
 
+test "zig fmt: line comment in array" {
+    try testTransform(
+        \\test "a" {
+        \\    var arr = [_]u32{
+        \\        0
+        \\        // 1,
+        \\        // 2,
+        \\    };
+        \\}
+        \\
+    ,
+        \\test "a" {
+        \\    var arr = [_]u32{
+        \\        0, // 1,
```

Comment:
I don't agree that this transformation — moving a next-line comment to a line-ending comment — is desirable or expected. It's probably more desirable than deleting the comment entirely, but this feels like a half-fix.

I'll comment on https://github.com/ziglang/zig/issues/2658 to make sure everyone is on the same page about expected output.

Reply:
The comment is moved only because the previous line doesn't have a trailing comma.
```zig
var arr = [_]u32{
    0,
    // 1
};
```
Will not move the comment.

Reply:
The case you just mentioned is not tested in this PR. It is arguably more common and important than the tests that are currently present in this PR. In #2658, I originally posted an example that has a trailing comma before comments on the next lines.

Reply:
I added test cases with trailing commas.

Reply:
Thanks for updating the test @Vexu 👍